databricks/koalas/plot.py

```diff
@@ -1186,181 +1185,170 @@ def _get_plot_backend(backend=None):
+
+        for arg, def_val in positional_args:
+            # map the argument if possible
+            if backend_name != "databricks.koalas.plot":
+                if backend_name in args_map:
+                    if arg in kwargs and arg in args_map[backend_name]:
+                        kwargs[args_map[backend_name][arg]] = kwargs.pop(arg)
+                # remove argument is default and not mapped
+                if arg in kwargs and kwargs[arg] == def_val:
+                    kwargs.pop(arg, None)
+            else:
+                if arg not in kwargs:
+                    kwargs[arg] = def_val
+
+        if backend_name != "databricks.koalas.plot":
+            data = data_preprocessor_map[kind](data)
+
+        return data, kwargs
+
+    def __call__(self, kind="line", backend="matplotlib", **kwargs):
+
         positional_args = locals()
-        plot_backend = _get_plot_backend(kwds.pop("backend", None))
+        plot_backend = _get_plot_backend(backend)
+        args = {**positional_args, **kwargs}
         # when using another backend, let the backend take the charge
+        plot_data, kwds = self._format_args(plot_backend.__name__, self.data, kind, args)
+
         if plot_backend.__name__ != "databricks.koalas.plot":
-            args = {**positional_args, **kwds}
-            plot_data, kwds = _get_args_map(plot_backend.__name__, self.data, kind, args)
             return plot_backend.plot(plot_data, kind=kind, **kwds)
 
-        return plot_series(
-            self.data,
-            kind=kind,
-            ax=ax,
-            figsize=figsize,
-            use_index=use_index,
-            title=title,
-            grid=grid,
-            legend=legend,
-            style=style,
-            logx=logx,
-            logy=logy,
-            loglog=loglog,
-            xticks=xticks,
-            yticks=yticks,
-            xlim=xlim,
-            ylim=ylim,
-            rot=rot,
-            fontsize=fontsize,
-            colormap=colormap,
-            table=table,
-            yerr=yerr,
-            xerr=xerr,
-            label=label,
-            secondary_y=secondary_y,
-            **kwds
-        )
+        if kind not in self._all_kinds:
+            raise ValueError(f"{kind} is not a valid plot kind")
+
+        from databricks.koalas import DataFrame, Series
 
+        if isinstance(self.data, Series):
+            if kind not in self._series_kinds:
+                return unsupported_function(class_name="pd.Series", method_name=kind)()
+            return plot_series(data=self.data, kind=kind, **kwds)
+        elif isinstance(self.data, DataFrame):
+            if kind not in self._dataframe_kinds:
+                return unsupported_function(class_name="pd.DataFrame", method_name=kind)()
+            return plot_frame(data=self.data, kind=kind, **kwds)
+
+    # added this, as it was present before
+    __call__.__doc__ = plot_frame.__doc__
     __call__.__doc__ = plot_series.__doc__
```

Comment:
This is just overwriting the exiting doc.

Reply:
I did not understand its use initially. Now I do. Instead of adding any particular functions documentation should I add doc string to `__call__` method?